### DISCUSSION TITLE
Is mitmproxy the right tool for stripping all images from traffic?

### DISCUSSION THREAD

**Discussion by rsend** [mitmproxy/mitmproxy#6161] [General]
Hi,
I haven't *tried* anything yet - but I'm looking for a tool that can help me strip all images from web traffic. The purpose of this is to reduce the amount of data sent to a remote site which is accessed over a VERY slow link (think dialup).

Can mitmproxy implement a filter like this?  A cursory search doesn't reveal anything, but this feels like the right tool for the job.

I'm currently accessing the internet with a Raspberry Pi Zero, all it does is forward the traffic to a wireless radio modem so ideally I can put something in-between more or less "as-is".

Thanks,
R

**Comment by mhils** [comment]
You can definitely write a small addon that just swaps out all images with
empty placeholders. I would do that over just intercepting these requests
to avoid infinite loading bars. :)

**Reply by rsend** [reply]
Cool, will look into it. First I'm having install issues... "Illegal instruction" trying to run after installing using pipx on a Raspberry Pi Zero :(
